### PR TITLE
Interactive

### PR DESCRIPTION
**Pull request by AbrahaKahsay**
## Project description
In this project, I will add some functionality to my application to make it interactive. The user will also be able to mark task completion by selecting the corresponding checkbox (or undo it by unchecking the checkbox). The updated tasks list will be stored in local storage.

**The requirements for this project are:**

- Add a new JavaScript file and import it as a module:
 it will contain methods related to the status updates (completed: true/false).
- Add event listener to the checkbox (change).
- Update items object's value for completed key upon user actions.
- Implement a function for the "Clear all completed" button (use filter() method).
- Store the updated array of items in local storage, so the user gets the correct list values after the page reloads.
## Built with
- Major languages: HTML, CSS, JavaScript
- Technologies used: VS Code, Github, Linters, Firefox web-browser
- webpack

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -61,10 +63,74 @@ const updateTask = () => {
...
+      list.style.textDecoration = 'line-through';
+    } else if (collection.array[i].complete === true) {
+      collection.array[i].complete = false;
+      localStorage.setItem('Tasks', JSON.stringify(collection.array));
+
+      const lists = document.getElementById(`list${i}`);
+      lists.style.textDecoration = 'none';
+    }
+  });
+}
```
Comment (medaminedev66) [inline src/index.js:93]: - [ ] The user should get the correct list values after the page reloads. Please make sure when you reload the page the checkboxes next to the tasks are checked if the tasks are completed.

File: src/index.js [hunk 2]
```diff
@@ -61,10 +63,74 @@ const updateTask = () => {
...
+  }
+  // store filtered items into local storage
+  localStorage.setItem('Tasks', stringData);
+
+  const remove = document.querySelector('.trashIcon');
+  remove.style.display = 'none';
+
+  updateTask();
+  updateIndex();
+});
```
Comment (medaminedev66) [inline src/index.js]: - [ ] The clear button does not work as expected. Only completed tasks  (their completed value is true) should be deleted when the Clear all completed button is clicked.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @AbrahaKahsay  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- There are no linter errors
- You have a descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (AbrahaKahsay) [reply src/index.js]: @medaminedev66 Thanks for the review. I couldn't figure out how to make the checkboxes stay the same before and after page refresh. could you please give me some insight on that? Thanks again!
I used the following code to save the checkboxes but not working.
image

**Review by Reem-lab — CHANGES_REQUESTED** [review]
## Hi  @AbrahaKahsay, 👋🏻👋🏻 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Descriptive pull request ✅✅
- Good commit messages ✅✅
- No Linters error ✅✅

 ## Required Changes ♻️
_Check the comments under the review._
[ REQUIRED ] Please make the changes that the previous code reviewed asked for the button
 to clear all the tasks and the checkbox.

[ HINT] you can first use a filter to specify the completed tasks then make for each to the filtered array and use the delete by one task method like this

```
const arrayOfTasks1 = arrayOfTasks.filter((task) => task.completed);
  arrayOfTasks1.forEach((task) => {
    deleteTaskWith(task.id);
  });
```

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AbrahaKahsay** [comment]
@Reem-lab Thanks for the feedback. It was helpful.

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @AbrahaKahsay 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**